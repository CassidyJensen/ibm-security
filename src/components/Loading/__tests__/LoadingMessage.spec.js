/**
 * @file Loading tests.
 * @copyright IBM Security 2018
 */

import { shallow } from 'enzyme';
import React from 'react';

import { LoadingMessage } from '../';

describe('LoadingMessage', () => {
  let loading;

  beforeEach(() => {
    loading = shallow(
      <LoadingMessage active className="" withOverlay small={false} />
    );
  });

  it('renders correctly', () => expect(loading).toMatchSnapshot());

  it('renders active=false correctly', () => {
    loading.setProps({ active: false });
    expect(loading).toMatchSnapshot();
  });

  it('renders withOverlay=false correctly', () => {
    loading.setProps({ withOverlay: false });
    expect(loading).toMatchSnapshot();
  });
});
