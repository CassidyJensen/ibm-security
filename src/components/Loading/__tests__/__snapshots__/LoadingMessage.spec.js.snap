// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingMessage renders active=false correctly 1`] = `
<div
  className="bx--loading-overlay bx--loading-overlay--stop"
>
  <Loading
    active={false}
    description="Active loading indicator"
    small={false}
    withOverlay={false}
  />
</div>
`;

exports[`LoadingMessage renders correctly 1`] = `
<div
  className="bx--loading-overlay"
>
  <Loading
    active={true}
    description="Active loading indicator"
    small={false}
    withOverlay={false}
  />
</div>
`;

exports[`LoadingMessage renders withOverlay=false correctly 1`] = `
<Loading
  active={true}
  description="Active loading indicator"
  small={false}
  withOverlay={false}
/>
`;
